src/item.cpp

```diff
@@ -4132,7 +4132,7 @@ bool item::gunmod_compatible( const item& mod, bool alert ) const
     } else if( get_free_mod_locations( mod.type->gunmod->location ) <= 0 ) {
         msg = string_format( _( "Your %1$s doesn't have enough room for another %2$s mod." ), tname().c_str(), _( mod.type->gunmod->location.c_str() ) );
 
-    } else if( ammo_remaining() > 0 || magazine_current() ) {
+    } else if( ( ammo_remaining() > 0 || magazine_current() ) && ammo_current() != "generic_no_ammo" ) {
```

Comment:
In the version with this modification in it? Because it doesn't fail with this line altered. I'll be happy to give you screenshots of anything that will help.